### PR TITLE
Add signal_blocker

### PR DESCRIPTION
**Pull request by lhamowski**

### CODE REVIEW

File: include/circle/reactive/signal.hpp [hunk 1]
```diff
@@ -459,6 +477,43 @@ class scoped_connection
...
+class signal_blocker
+{
+public:
+    template <typename... Args>
+    signal_blocker(signal<Args...>& s) noexcept : connections_{s.connections_}
+    {
+        auto& connections = s.connections_;
+        const auto ids = connections->ids();
+        if (ids.empty())
+            CIRCLE_WARN("Creating signal_blocker without any connections");
```
Comment (jm4R) [inline include/circle/reactive/signal.hpp]: I don't think it is something we should warn about

File: include/circle/reactive/signal.hpp [hunk 2]
```diff
@@ -459,6 +477,43 @@ class scoped_connection
...
+        if (auto connections = connections_.lock())
+        {
+            for (auto id : blocked_)
+                connections->block(id, false);
+        }
+    }
+
+private:
+    std::weak_ptr<detail::connections_container_base> connections_;
+    detail::id_set blocked_;
```
Comment (jm4R) [inline include/circle/reactive/signal.hpp]: More straightforward solution than constructing `unordered_set` would be just introducing `bool connection_container::blocked` property. What do you think?

**Review by jm4R — CHANGES_REQUESTED** [review]
Thanks for the valuable contribution. The feature is handy but I feel we can make the implementation lighter. Please see my comments.

File: include/circle/reactive/signal.hpp (hunk 2 above)
Comment (lhamowski) [reply include/circle/reactive/signal.hpp]: You're absolutely right

**Review by jm4R — APPROVED** [review]

**Pull request merged**